### PR TITLE
refactor(core): remove deprecated public APIs

### PR DESCRIPTION
**Pull request by idoros**
This PR removes code that is no longer required due to exports being removed from `@stylable/core` index

- remove `index-deprecated` exports
- rewire stylable inter-package internals through `index-internal`
- delete unused code
- remove exports where the export was only required for the main index exports
- remove code marked with `@deprecated` (that can be removed now)

> **CHANGE**: Part of the code removed included `SRule` fields (all except `stScopeSelector` that will be removed once AST mods will move to transformer). The `SRule.mixin` removed the mechanism that reported some of mixin diagnostics during the process phase, and now they are reported during the transform phase.

### CODE REVIEW

File: packages/core/src/parser.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ import postcss, { ProcessOptions, Root, parse as cssParse } from 'postcss';
 import postcssNested from 'postcss-nested';
 import postcssJS from 'postcss-js';
 import safeParser from 'postcss-safe-parser';
-import type { CSSObject } from './types';
+
+export type CSSObject = any & object;
```
[the remark this answers is not in the document]
Comment (barak007) [reply packages/core/src/parser.ts:6]: there are now good types for it

File: packages/core/src/stylable-processor.ts [hunk 2]
```diff
@@ -425,8 +421,8 @@ export class StylableProcessor implements FeatureContext {
             } else {
                 this.diagnostics.warn(decl, processorWarnings.CANNOT_EXTEND_IN_COMPLEX());
             }
-        } else if (decl.prop === STMixin.MixinType.ALL || decl.prop === STMixin.MixinType.PARTIAL) {
-            STMixin.hooks.analyzeDeclaration({ context: this, decl });
+            // } else if (decl.prop === STMixin.MixinType.ALL || decl.prop === STMixin.MixinType.PARTIAL) {
```
Comment (barak007) [inline packages/core/src/stylable-processor.ts:424]: remove comments

**Review by barak007 — COMMENTED** [review]
Amazing  Amazing Amazing work!

File: packages/core/src/parser.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (idoros) [reply packages/core/src/parser.ts:6]: its still exported from the inner module, just not from the public index. The types can be replaced in a later PR.

File: packages/core/src/stylable.ts [hunk 3]
```diff
@@ -14,7 +14,6 @@ import {
 } from './stylable-transformer';
 import type { IStylableOptimizer, ModuleResolver } from './types';
 import { createDefaultResolver } from './module-resolver';
-import { warnOnce } from './helpers/deprecation';
 import { STVar, CSSCustomProperty } from './features';
 import { Dependency, visitMetaCSSDependencies } from './visit-meta-css-dependencies';
 import * as postcss from 'postcss';
```
[the remark this answers is not in the document]
Comment (idoros) [reply packages/core/src/stylable.ts:19]: removed everything

**Pull request merged**